### ISSUE TITLE
Feature: Improve default grouping of sort-classes

### ISSUE THREAD

**Issue by difosfor** [azat-io/eslint-plugin-perfectionist#309]
### What rule do you want to change?

sort-classes

### Describe the problem

By default:

- Public members are sorted below protected and private members instead of above
- Property get and set methods are sorted along with methods instead of properties
- Function properties, e.g: bound methods, aren't sorted along with methods

I've improved the groups configuration as follows to fix this:

### Code example

```js
groups: [
  'index-signature',
  'static-property',
  'protected-static-property',
  'private-static-property',
  'static-block',
  ['property', 'accessor-property', 'get-method', 'set-method'],
  ['protected-property', 'protected-accessor-property', 'protected-get-method', 'protected-set-method'],
  ['private-property', 'private-accessor-property', 'private-get-method', 'private-set-method'],
  'constructor',
  ['static-method', 'static-function-property'],
  ['protected-static-method', 'protected-static-function-property'],
  ['private-static-method', 'private-static-function-property'],
  ['method', 'function-property'],
  ['protected-method', 'protected-function-property'],
  ['private-method', 'private-function-property'],
  'unknown',
]
```

### Additional comments

I think these would provide more sensible defaults than the current?

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hi! This issue is similar to https://github.com/azat-io/eslint-plugin-perfectionist/issues/236.

There is a discussion inside where we debated whether we should change the default configuration or not by regrouping function properties with methods.

I chose not to submit a PR, feeling that we were venturing into a grey area where the changes could be viewed as overly subjective.

**Comment by hugop95** [comment]
I personally agree with all of your proposals, this is in fact exactly what I use.

The tricky part is indeed prevent breaking changes. Note that we **did** change the default configuration before: first when then new framework for `sort-classes` was released, because many class members were detected as `unknown` and placed at the bottom, then when we decided to put `static-block` at the top (due to some potential dependency issues at the time).

So it's not like such things were never done outside of major releases, but it's best to avoid it as much as possible. I like the idea of adding that in the next major release, however. It's going to depend on what @azat-io plans on adding in v4.

**Comment by hugop95** [comment]
Actually, timing a new major release with https://github.com/azat-io/eslint-plugin-perfectionist/pull/304 could be quite convenient, this allows us to improve the default configuration as well properly.
Furthermore, I think that the new `sort-decorators` rule should ideally be in major version, as it can unexpectedly affect users who heavily rely on annotations.

**Comment by difosfor** [comment]
Looks like this can be closed as implemented by #320, thanks @hugop95!

Looking forward to the next version of Biome being released; I'll just patch the config as in your PR for now.
